homeassistant/components/sensor/seventeentrack.py

```diff
@@ -0,0 +1,274 @@
+NOTIFICATION_DELIVERED_URL_SCAFFOLD = 'https://t.17track.net/track#nums={0}'
+
+VALUE_DELIVERED = 'Delivered'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Optional(CONF_SHOW_ARCHIVED, default=False): cv.boolean,
+    vol.Optional(CONF_SHOW_DELIVERED, default=False): cv.boolean,
+})
+
+
+async def async_setup_platform(
+        hass, config, async_add_entities, discovery_info=None):
+    """Configure the platform and add the sensors."""
+    from py17track import Client
+    from py17track.errors import SeventeenTrackError
+
+    websession = aiohttp_client.async_get_clientsession(hass)
+
+    client = Client(websession)
+
+    try:
+        login_result = await client.profile.login(
+            config[CONF_USERNAME], config[CONF_PASSWORD])
+
+        if not login_result:
+            _LOGGER.error('Invalid username and password provided')
+            return
+    except SeventeenTrackError as err:
+        _LOGGER.error('There was an error while logging in: %s', err)
+        return
+
+    scan_interval = config.get(CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL)
+
+    data = SeventeenTrackData(
+        client, async_add_entities, scan_interval, config[CONF_SHOW_ARCHIVED],
+        config[CONF_SHOW_DELIVERED])
+    await data.async_update()
+
+    sensors = []
+
+    for status, quantity in data.summary.items():
+        sensors.append(SeventeenTrackSummarySensor(data, status, quantity))
+
+    for package in data.packages:
+        sensors.append(SeventeenTrackPackageSensor(data, package))
+
+    async_add_entities(sensors, True)
+
+
+class SeventeenTrackSummarySensor(Entity):
+    """Define a summary sensor."""
+
+    def __init__(self, data, status, initial_state):
+        """Initialize."""
+        self._attrs = {ATTR_ATTRIBUTION: DEFAULT_ATTRIBUTION}
+        self._data = data
+        self._state = initial_state
+        self._status = status
+
+    @property
+    def device_state_attributes(self):
+        """Return the device state attributes."""
+        return self._attrs
+
+    @property
+    def icon(self):
+        """Return the icon."""
+        return 'mdi:package'
+
+    @property
+    def name(self):
+        """Return the name."""
+        return 'Packages {0}'.format(self._status)
+
+    @property
+    def state(self):
+        """Return the state."""
+        return self._state
+
+    @property
+    def unique_id(self):
+        """Return a unique, HASS-friendly identifier for this entity."""
+        return 'summary_{0}_{1}'.format(
+            self._data.account_id, slugify(self._status))
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit the value is expressed in."""
+        return 'packages'
+
+    async def async_update(self):
+        """Update the sensor."""
+        await self._data.async_update()
+
+        self._state = self._data.summary[self._status]
```

Comment:
If there's an error with the data update, `self._data.summary` is set to empty dict, and we will get a `KeyError` here.

We should mark entity as unavailable instead.

Reply:
I went about this slightly differently in 61a19d7 – let me know what you think (and if I missed something).